crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -888,3 +890,18 @@ impl<S> EventHandler<S> {
             .expect("Mutex to not be poisoned")
     }
 }
+
+#[async_trait]
+pub trait EventHandlerTrait: Send + Sync {
```

Comment:
I am bit saddened that we have to introduce another abstraction, but I guess there's no better way to do this if we don't tear down the `ln-dlc-node` crate.

Reply:
I thought about this quite a bit and came to the conclusion that this indeed is the lesser evil. 
Not a big trade-off really for reusing a lot of the code.

if we ever have time to look at this again, I'd be keen to see where you push it - at least it should be easier to refactor from this point :)